### DISCUSSION TITLE
Disassembly from a .NET Core 6.0 app

### DISCUSSION THREAD

**Discussion by pdavis68** [icsharpcode/ILSpy#2727] [General] [answered]
I've got a .NET core app in 6.0. My code is very simple (and I may be doing stuff completely wrong. If there are docs on this stuff, a pointer to them would be great).  The code is adapted from a snippet I found in a google search.

```
        public static string DisassembleAssembly(this string assemblyName)
        {
            var plainTextOutput = new PlainTextOutput();
            var decompiler = new CSharpLanguage();
            var al = new AssemblyList("Pedro");
            var asm = al.OpenAssembly(assemblyName, false);
            decompiler.DecompileAssembly(asm, plainTextOutput, new DecompilationOptions() { FullDecompilation = true });
            return Regex.Replace(Regex.Replace(plainTextOutput.ToString(), @"\n|\r", " "), @"\s+", " ");
        }

```
When it hits the OpenAssembly call, it gave me an error that it couldn't find PresentationFramework 4.0. So I added the following to my .csproj:

```
    <Reference Include="PresentationFramework">
     <HintPath>C:\Windows\Microsoft.Net\assembly\GAC_MSIL\PresentationFramework\v4.0_4.0.0.0__31bf3856ad364e35\PresentationFramework.dll</HintPath>
    </Reference>

```
I reran it and that went away, but then I got the same error for WindowsBase. I added a similar link for WindowsBase, but the error never went away.
```
    <Reference Include="WindowsBase">
      <HintPath>C:\Windows\Microsoft.Net\assembly\GAC_MSIL\WindowsBase\v4.0_4.0.0.0__31bf3856ad364e35\WindowsBase.dll</HintPath>
    </Reference>
```
My IL Spy version info:

ILSpy version 7.2.1.6856
.NET version 4.8.4488.0

Out of ideas. I'm sure there's a better way to do this. I'd love to learn it. Thanks!

**Comment by siegfriedpammer** [comment, marked as answer]
Are you trying to write a plugin for ILSpy? Or are you trying to use the decompiler in your own application?

**Reply by pdavis68** [reply]
I'm trying to decompile code in my own app.  I'm essentially trying to do something like [this](https://godbolt.org/), except for C#. 
So I have a pane with C# code. I compile it into a .DLL and then I'm trying to use ILSpy to disassemble it, to show that in the adjoining pane.

**Reply by siegfriedpammer** [reply]
So you are trying to create https://sharplab.io ?

**Reply by pdavis68** [reply]
LOL. Apparently I was.

**Reply by siegfriedpammer** [reply]
Our wiki https://github.com/icsharpcode/ILSpy/wiki/Getting-Started-With-ICSharpCode.Decompiler should help you to get started. If you have any further questions, please feel free to ask.

EDIT: Didn't see your response before... If you want to do something else with the decompiler engine, the link above might be helpful to you.

**Comment by dgrunwald** [comment]
When using the decompiler, use class CSharpDecompiler. There is no need to reference ILSpy.exe -- `AssemblyList` is our class for the ILSpy GUI and is not meant to be used as a library.
See also https://stackoverflow.com/questions/56893130/how-to-use-icsharpcode-decompiler-to-decompile-a-whole-assembly-to-a-text-file
